src/Symfony/Component/ExpressionLanguage/Node/ArgumentsNode.php

```diff
@@ -27,12 +27,14 @@ public function compile(Compiler $compiler)
 
     public function dump()
     {
-        $str = '';
+        $tokens = array();
 
         foreach ($this->getKeyValuePairs() as $pair) {
-            $str .= sprintf('%s, ', $pair['value']->dump());
+            $tokens[] = ', ';
+            $tokens[] = $pair['value'];
         }
+        array_shift($tokens);
```

Comment:
isn't it better to add the ``,  `` token after each pair, and to pop the last one here ? It would avoid having to reindex the array